### PR TITLE
[OP#45466]Remove expired direct upload tokens using background jobs

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR implements the background jobs to remove the expired direct upload tokens from the databases.

### Related Workpackage
https://community.openproject.org/projects/817/work_packages/45466/activity

### CODE REVIEW

File: lib/Service/DatabaseService.php [hunk 1]
```diff
@@ -102,6 +103,27 @@ public function getTokenInfoFromDB(string $token): array {
...
+	 *
+	 * Get All token information from database
+	 *
+	 * @throws Exception
+	 */
+	public function deleteAllExpiredTokenFromDB(): void {
+		$query = $this->db->getQueryBuilder();
+		$query->select('token', 'expires_on')
+			->from($this->table);
+		$req = $query->executeQuery();
```
Comment (individual-it) [inline lib/Service/DatabaseService.php:116]: just make a single request and let the DB deal with it, that will be much more performant that doing multiple requests.
MySQL should work like this: `DELETE FROM oc_direct_upload WHERE  expires_on < UNIX_TIMESTAMP();`

File: lib/Service/DirectUploadService.php [hunk 2]
```diff
@@ -56,7 +56,6 @@ class DirectUploadService {
 
 	/** @var string time of token expiration */
 	private string $expiryTime = '+1 hour';
-
```
Comment (individual-it) [inline lib/Service/DirectUploadService.php:59]: not needed

File: lib/Service/DatabaseService.php [hunk 3]
```diff
@@ -102,6 +103,27 @@ public function getTokenInfoFromDB(string $token): array {
 		];
 	}
 
+	/**
+	 *
+	 * Get All token information from database
```
Comment (individual-it) [inline lib/Service/DatabaseService.php:108]: not true, be honest ;-)

File: lib/Service/DatabaseService.php [hunk 4]
```diff
@@ -102,6 +103,27 @@ public function getTokenInfoFromDB(string $token): array {
...
+	 *
+	 * @throws Exception
+	 */
+	public function deleteAllExpiredTokenFromDB(): void {
```
Comment (individual-it) [inline lib/Service/DatabaseService.php:112]:
```suggestion
	public function deleteExpiredTokens(): void {
```
as this is a DB class, its clear that its deleting from the DB

**Review by individual-it — CHANGES_REQUESTED** [review]

File: lib/Service/DatabaseService.php (hunk 1 above)
Comment (individual-it) [reply lib/Service/DatabaseService.php:116]: for pgsql it might be easier to generate the timestamp in PHP. But still just have a single delete request

Comment (SagarGi) [reply lib/Service/DatabaseService.php]: @individual-it do we want to use `UNIX_TIMESTAMP();` function which is for mysql. why not using the date instead? does it not work for pgsql?

**Comment by individual-it** [comment]
The app is not enabled in CI when we run the unit tests. I would suggest to merge #323, rebase and then adjust then adjust the CI steps

**Review by individual-it — APPROVED** [review]

**Comment by SagarGi** [comment]
> The app is not enabled in CI when we run the unit tests. I would suggest to merge #323, rebase and then adjust then adjust the CI steps

sure !!

**Review by SwikritiT — APPROVED** [review]

**Pull request merged**